sdk/core/azure-core-amqp/inc/azure/core/amqp/management.hpp

```diff
@@ -34,7 +35,7 @@ namespace Azure { namespace Core { namespace Amqp { namespace _internal {
     Cancelled,
   };
 
-  struct ManagementOptions
+  struct ManagementClientOptions
```

Comment:
If we ever want to allow customers to mock our types (which is a pretty hard requirement for customers that use test driven development), we are going to have to reconsider the use of final in our types.

But since our current stance is that we don't allow customers to mock SDK objects, it can stand.

Reply:
For deriving while testing, we currently have an approach for that - see https://github.com/Azure/azure-sdk-for-cpp/blob/fd687c32fa18bd8f1daacadd8c8aff870eb08e26/sdk/core/azure-core/inc/azure/core/http/policies/policy.hpp#L366-L368